wallet-new/test/MarshallingSpec.hs

```diff
@@ -5,28 +5,62 @@ import           Universum
 import           Cardano.Wallet.API.V1.Types
 import           Cardano.Wallet.Orphans ()
 import           Data.Aeson
-import           Pos.Util.BackupPhrase (BackupPhrase)
+import           Data.Typeable (typeRep)
+import           Pos.Crypto (PassPhrase, emptyPassphrase)
 import           Test.Hspec
 import           Test.Hspec.QuickCheck
-import           Test.QuickCheck
+import           Test.QuickCheck (Arbitrary, arbitrary, forAll, (===))
 import           Test.QuickCheck.Instances ()
+import           Test.QuickCheck.Property (Property, failed, property, reason)
+
+import           Pos.Util.BackupPhrase (BackupPhrase)
 
 import qualified Pos.Core.Types as Core
 
 -- | Tests whether or not some instances (JSON, Bi, etc) roundtrips.
 spec :: Spec
 spec = describe "Marshalling & Unmarshalling" $ do
-  prop "Aeson BackupPhrase roundtrips" (aesonRoundtrip @BackupPhrase Proxy)
-  prop "Aeson AssuranceLevel roundtrips" (aesonRoundtrip @AssuranceLevel Proxy)
-  prop "Aeson Payment roundtrips" (aesonRoundtrip @Payment Proxy)
-  prop "Aeson PaymentDistribution roundtrips" (aesonRoundtrip @PaymentDistribution Proxy)
-  prop "Aeson NewWallet roundtrips" (aesonRoundtrip @NewWallet Proxy)
-  prop "Aeson Coin roundtrips" (aesonRoundtrip @Core.Coin Proxy)
-  prop "Aeson TransactionGroupingPolicy roundtrips" (aesonRoundtrip @TransactionGroupingPolicy Proxy)
-  prop "Aeson TransactionType roundtrips" (aesonRoundtrip @TransactionType Proxy)
-  prop "Aeson Transaction roundtrips" (aesonRoundtrip @Transaction Proxy)
+    describe "Roundtrips" $ do
+        aesonRoundtripProp @BackupPhrase Proxy
+        aesonRoundtripProp @AssuranceLevel Proxy
+        aesonRoundtripProp @Payment Proxy
+        aesonRoundtripProp @PaymentDistribution Proxy
+        aesonRoundtripProp @NewWallet Proxy
+        aesonRoundtripProp @Core.Coin Proxy
+        aesonRoundtripProp @TransactionGroupingPolicy Proxy
+        aesonRoundtripProp @TransactionType Proxy
+        aesonRoundtripProp @Transaction Proxy
+
+    describe "Special" $ do
+        describe "password" $ do
+            prop "empty" $
+                jsonString "" `decodesTo` (== emptyPassphrase)
+            prop "nonempty" $
+                jsonString (fromString $ replicate 64 'a')
+                    `decodesTo` (/= emptyPassphrase)
+            prop "wrong length" $
+                -- currently passphrase should be either empty or of length 32
+                decodingFails @PassPhrase "aabbcc" Proxy
 
 
 aesonRoundtrip :: (Arbitrary a, ToJSON a, FromJSON a, Eq a, Show a) => proxy a -> Property
 aesonRoundtrip (_ :: proxy a) = forAll arbitrary $ \(s :: a) -> do
     decode (encode (toJSON s)) === Just s
+
+aesonRoundtripProp
+    :: (Arbitrary a, ToJSON a, FromJSON a, Eq a, Show a, Typeable a)
+    => proxy a -> Spec
+aesonRoundtripProp proxy =
+    prop ("Aeson " <> show (typeRep proxy) <> " roundtrips") (aesonRoundtrip proxy)
+
+decodesTo :: FromJSON a => LByteString -> (a -> Bool) -> Property
```

Comment:
I think this is better written as a normal `hspec` `Expectation` rather than a `Property`.